### PR TITLE
Add support for streaming a log file to a specified path.

### PR DESCRIPTION
**Pull request by dwick**
:eyeglasses: @ajacksified @umbrae 

See [embeds pr](https://github.com/reddit/hivemind/pull/14) for usage.

### CODE REVIEW

File: src/data/stream.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+var crypto = require('crypto');
```
Comment (umbrae) [inline src/data/stream.js]: Unnecessary import?

**Comment by umbrae** [comment]
:fish: from my perspective but I only really started getting into hivemind deeply today.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "metron",
-  "version": "0.2.1",
+  "version": "0.3.1",
```
Comment (ajacksified) [inline package.json]: Should this be `0.3.0` instead of `0.3.1`?

File: src/metron.js [hunk 3]
```diff
@@ -180,6 +180,7 @@ Metron.prototype.processParameters = function(req, res) {
 Metron.dataAdapters = {
   Statsd: require('./data/statsd'),
   Log: require('./data/log'),
+  Stream: require('./data/stream'),
```
Comment (ajacksified) [inline src/metron.js]: Since this acts like a Log, we should probably call it `LogStream` instead.

**Comment by ajacksified** [comment]
Couple of minor things; looks good, and hooray for abstracting out the stream. :nail_care:

File: package.json (hunk 2 above)
Comment (dwick) [inline package.json]: yes

**Comment by dwick** [comment]
:haircut: 

Realized i had missed some changes but already renamed the file and it was too much of a pain to split out.  `defaultStreamConfig` was replaced by just overwriting `Log#save`.  Didn't see a big reason for that to be configurable.  Should we make `Log#save` just `console.log` too and use `LogStream` in hivemind?

**Comment by ajacksified** [comment]
Yeah - let's have hivemind use `LogStream` where it currently uses `Log`. :fish:

**Comment by dwick** [comment]
ok, updating hivemind to work with this now.  i'll link the PRs together

**Comment by dwick** [comment]
:haircut: required a couple more changes

**Comment by ajacksified** [comment]
:fish:

**Comment by umbrae** [comment]
:fish: 2

**Pull request merged**